### PR TITLE
Construct feedbacks url using tag slugs instead of ordinal numbers

### LINKED ISSUE #91: Changes to the feedback tag list have an odd impact on URLs
I’ll just explain how to reproduce this:

1. Use the feedback-filtering form to search for messages with a particular feedback tag. Here’s an example URL from the O1 course: 
https://jutut.cs.aalto.fi/manage/55/feedbacks/?contains_text=on&tags_1_10=I232&order_by=-timestamp
Make a bookmark with this URL. (Side note: I use that example tag to mark urgent stuff that I should deal with, so I have the above link bookmarked.)

2. Revisit the URL via the bookmark, for example. Notice how the selected tag is highlighted in the filtering form’s UI, as it should be.

3. Go to Manage Feedback tags and add some random feedback tag. Let’s say we add a tag called "aaaa" at the top of the list.

4. Go back to the above URL. Notice how the selected tag is *not* highlighted any more. 

5. Click the tag in the form again and search. Notice that the URL is now different than before the "aaaa" tag was added. (The "tags" parameter has changed. Presumably the added feedback tag impacted on the numbering of the other, existing tags.)

The same in simpler terms: whenever I add a feedback tag (or another staff member does), my Jutut bookmarks (partially) break.

### PR DESCRIPTION
**Pull request by Mikael-Lenander**
# Description

**What?**

When filtering feedback based on student and feedback tags, constructs the url parameters using tags slugs. Currently uses ordinal numbers.

**Why?**

With the current implementation, the url changes for the same query if new tags are added. This is not ideal.

**How?**

A new class UniqueColortagIEAndOrFilter is added that inherits from ColortagIEAndOrFilter (defined in django_colortag). The only modification is that self.widgets_names is contructed in a new way. A cleaner approach would probably be to modify the django_colortag library directly, but this may have unintended side effects.

Fixes #91

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Saved feedbacks url as a bookmark as described in #91. Added and deleted tags and checked that the bookmark still works.

**Did you test the changes in**

- [ ] Chrome
- [x] Firefox
- [ ] This pull request cannot be tested in the browser.

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

**Comment by etanttila** [comment]
I indeed would recommend modifying the django-colortag library, as having the slugs in the names rather than ordinal numbers would probably be useful for other uses as well, and it would reduce the amount of repetition (and copied code) significantly. The repetition also could make it more difficult in the future to make changes to the widgets.

In addition, currently the url queries always have an extra 1 in between the filter and tag names (e.g. `tags_1_bug=I1&tags_1_test2=I3&student_tags_1_local=I2`, I wonder how easy it would be to remove those.

I wonder if the subwidgets' names need to be set in the multiwidget, of if the solution could be as easy as setting the name of the `ColortagIncludeExcludeWidget` to be the name of the tag.

**Comment by ihalaij1** [comment]
Ella's suggestion does seem good.
Could you try to implement this in the django-colortag repository instead?

**Comment by Mikael-Lenander** [comment]
Thanks for the feedback. I created a new pr to the django-colortag repository.

**Comment by etanttila** [comment]
I'll close this PR, since it has changed so much, and with the new implementation, only the requirements file needs to be updated. This is implemented in #106
